test/plugin/test_out_elasticsearch.rb

```diff
@@ -1762,6 +1800,47 @@ def test_writes_to_default_index(data)
     assert_equal(index_name, index_cmds.first['index']['_index'])
   end
 
+  # gzip compress data
+  def gzip(string, strategy)
+    wio = StringIO.new("w")
+    w_gz = Zlib::GzipWriter.new(wio, strategy = strategy)
+    w_gz.write(string)
+    w_gz.close
+    wio.string
+  end
+
+
+  def test_writes_to_default_index_with_compression
+    config = %[
+      compression_level default_compression
+    ]
+
+    bodystr = %({
+          "took" : 500,
+          "errors" : false,
+          "items" : [
+            {
+              "create": {
+                "_index" : "fluentd",
+                "_type"  : "fluentd"
+              }
+            }
+           ]
+        })
+
+    compressed_body = gzip(bodystr, Zlib::DEFAULT_COMPRESSION)
+
+    elastic_request = stub_request(:post, "http://localhost:9200/_bulk").
+        to_return(:status => 200, :headers => {"Content-Type": "Application/json"}, :body => compressed_body)
```

Comment:
Could you use rocket hash style?
ES plugin still support Ruby 2.1.
JSON like Hash style had been supported since Ruby 2.2

Reply:
fixed